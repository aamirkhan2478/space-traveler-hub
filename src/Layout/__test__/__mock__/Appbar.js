import React from 'react';
import { BrowserRouter, NavLink } from 'react-router-dom';
import { Container, Nav, Navbar } from 'react-bootstrap';
import Logo from '../../assets/Images/planet.png';

const Appbar = () => (
  <BrowserRouter>
    <Navbar collapseOnSelect expand="lg" bg="light" variant="light">
      <Container>
        <Navbar.Brand href="/">
          <img src={Logo} alt="logo" className="img-fluid image-size" />
          Space Traveler`&apos;s Hub
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
        <Navbar.Collapse id="responsive-navbar-nav">
          <Nav className="ms-auto">
            <Nav.Link>
              <NavLink to="/">Rockets</NavLink>
            </Nav.Link>
            <Nav.Link>
              <NavLink to="/mission">Missions</NavLink>
            </Nav.Link>
            <Nav.Link>
              <NavLink to="/profile">My Profile</NavLink>
            </Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  </BrowserRouter>
);

export default Appbar;
