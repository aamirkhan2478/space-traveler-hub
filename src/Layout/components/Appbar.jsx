import React from 'react';
import { Container, Nav, Navbar } from 'react-bootstrap';
import { NavLink, useLocation } from 'react-router-dom';
import Logo from '../assets/Images/planet.png';
import '../assets/Styles/navbar.css';

const Appbar = () => {
  const location = useLocation();

  return (
    <Navbar collapseOnSelect expand="lg" bg="light" variant="light">
      <Container>
        <Navbar.Brand href="/">
          <img src={Logo} alt="logo" className="img-fluid image-size" />
          Space Traveler`&apos;s Hub
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
        <Navbar.Collapse id="responsive-navbar-nav">
          <Nav className="ms-auto">
            <Nav.Link>
              <NavLink
                to="/"
                className={
                  location.pathname === '/'
                    ? 'nav-link active-link text-primary'
                    : 'nav-link text-primary'
                }
              >
                Rockets
              </NavLink>
            </Nav.Link>
            <Nav.Link>
              <NavLink
                to="/mission"
                className={
                  location.pathname === '/mission'
                    ? 'nav-link active-link text-primary'
                    : 'nav-link text-primary'
                }
              >
                Missions
              </NavLink>
            </Nav.Link>
            <Nav.Link>
              <NavLink
                to="/profile"
                className={
                  location.pathname === '/profile'
                    ? 'nav-link active-link text-primary'
                    : 'nav-link text-primary'
                }
              >
                My Profile
              </NavLink>
            </Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};

export default Appbar;
