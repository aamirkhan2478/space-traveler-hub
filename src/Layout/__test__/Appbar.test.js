import renderer from 'react-test-renderer';
import Appbar from './__mock__/Appbar';

it('renders navbar correctly', () => {
  const tree = renderer.create(<Appbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
