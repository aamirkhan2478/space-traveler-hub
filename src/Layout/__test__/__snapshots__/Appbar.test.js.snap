// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navbar correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt="logo"
        className="img-fluid image-size"
        src="planet.png"
      />
      Space Traveler\`'s Hub
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="ms-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rr-ui-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Rockets
          </a>
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <a
            className=""
            href="/mission"
            onClick={[Function]}
          >
            Missions
          </a>
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <a
            className=""
            href="/profile"
            onClick={[Function]}
          >
            My Profile
          </a>
        </a>
      </div>
    </div>
  </div>
</nav>
`;
